### ISSUE TITLE
Media type spoofing

### ISSUE THREAD

**Issue by alolis** [musaffa/file_validators#1]
Hello,

I think that media type spoofing is a bit problematic. For example I have a my_file.torrent which has a content type application/x-bittorrent and the file, even though is valid, it is being detected as spoofed.

I believe that the validation shouldn't make such an automatic detection but it should be up to the caller, via a new validator called 'file_extension_validation' to make such decisions.

Thank you for your time

**Comment by musaffa** [comment]
I will test with torrent files and see what happens. What content type is detected for your my_file.torrent? It should work if it is 'application/***'.

**Comment by alolis** [comment]
The content type is `application/x-bittorrent`

**Comment by alolis** [comment]
I think this is a more generic problem because I am having the same problem with `application/octet-stream`

**Comment by alolis** [comment]
What exactly is the validator expecting? The params[:my_file] object?

**Comment by musaffa** [comment]
It doesn't detect a valid torrent file as spoofed in my development machine. It also works with the integration test of this gem.
Perhaps your app is showing its application specific behaviour. To debug the behaviour, set break points on [spoof detector](https://github.com/musaffa/file_validators/blob/master/lib/file_validators/utils/media_type_spoof_detector.rb#L33) to see what's happening there.

What operating system you're using?

**Comment by alolis** [comment]
I am on Yosemite. I will do the debug and let you know.

One question though; What are you passing to the validator? The `params[:my_file]` object or something else?

**Comment by musaffa** [comment]
If you are uploading for the first time, the form automatically creates a tempfile object. It is then sent as a [value](https://github.com/musaffa/file_validators/blob/master/lib/file_validators/validators/file_content_type_validator.rb#L15) in the content type detector.

Run this command below from Yosemite shell:
```shell
file -b --mime-type path_to_the_torrent_file
```
A valid torrent file should return `application/x-bittorrent`

**Comment by alolis** [comment]
I can see the tempfile inside `params[:my_file].tempfile`. Is that what I have to use the validator with or `params[:my_file]` will do it?

I run the function and it returns `application/x-bittorrent` on my test torrent

**Comment by alolis** [comment]
OK...it seems that I must pass params[:my_file].tempfile. Maybe you should add this to the README so other silly people like me to see it :)

**Comment by musaffa** [comment]
It's ok. As your problem has been solved, I'm closing this issue.
